Apps/Examples/Examples/All Examples/Annotations/PointAnnotationClusteringExample.swift

```diff
@@ -0,0 +1,90 @@
+        // The image named `fire-station-11` is included in the app's Assets.xcassets bundle.
+        let image = UIImage(named: "fire-station-11")!
+        // Fire_Hydrants.geojson contains information about fire hydrants in the District of Columbia.
+        // It was downloaded on 6/10/21 from https://opendata.dc.gov/datasets/DCGIS::fire-hydrants/about
+        // Decode the GeoJSON into a feature collection
+        _ = Bundle.main.url(forResource: "Fire_Hydrants", withExtension: "geojson")!
+        guard let featureCollection = try? decodeGeoJSON(from: "Fire_Hydrants") else {
+            return
+        }
+
+        // Create an array of annotations for each fire hydrant
+        var annotations = [PointAnnotation]()
+        for feature in featureCollection.features {
+            guard let geometry = feature.geometry, case let Geometry.point(point) = geometry else {
+                return
+            }
+            var pointAnnotation = PointAnnotation(coordinate: point.coordinates)
+            pointAnnotation.image = .init(image: image, name: "fire-station-11")
+            annotations.append(pointAnnotation)
+        }
+
+        createClusters(annotations: annotations)
+    }
+
+    func createClusters(annotations: [PointAnnotation]) {
+        // Use a step expressions (https://docs.mapbox.com/mapbox-gl-js/style-spec/#expressions-step)
+        // with three steps to implement three sizes of circles:
+        //   * 15 when point count is less than 50
+        //   * 20 when point count is between 50 and 100
+        //   * 25 when point count is greater than or equal to 100
+        let circleRadiusExpression = Exp(.step) {
+            Exp(.get) {"point_count"}
+            15
+            50
+            20
+            100
+            25
+        }
+
+        // Select the options for clustering and pass them to the PointAnnotationManager to display
+        let clusterOptions = ClusterOptions(clusterRadius: 75, circleRadius: .expression(circleRadiusExpression), textColor: .constant(StyleColor(#colorLiteral(red: 0, green: 0, blue: 0, alpha: 1))), colorLevels: [(100, StyleColor(.red)), (50, StyleColor(.cyan)), (0, StyleColor(.green))])
+        let pointAnnotationManager = mapView.annotations.makePointAnnotationManager(clusterOptions: clusterOptions)
+        pointAnnotationManager.annotations = annotations
+
+        finish()
+    }
+
+    // Load GeoJSON file from local bundle and decode into a `FeatureCollection`.
+    func decodeGeoJSON(from fileName: String) throws -> FeatureCollection? {
+        guard let path = Bundle.main.path(forResource: fileName, ofType: "geojson") else {
+            preconditionFailure("File '\(fileName)' not found.")
+        }
+        let filePath = URL(fileURLWithPath: path)
+        var featureCollection: FeatureCollection?
+        do {
+            let data = try Data(contentsOf: filePath)
+            featureCollection = try JSONDecoder().decode(FeatureCollection.self, from: data)
```

Comment:
I have the calling function `addPointAnnotations()` offload the parsing and annotation array creation. Should I switch so that only parsing is offloaded from the main queue?